frontend/src/app/AppNotificationDrawer.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  Button,
+  ButtonVariant,
+  NotificationDrawer,
+  NotificationDrawerHeader,
+  NotificationDrawerBody,
+  NotificationDrawerList,
+  NotificationDrawerListItem,
+  NotificationDrawerListItemHeader,
+  NotificationDrawerListItemBody,
+  EmptyStateVariant,
+  EmptyState,
+  EmptyStateBody,
+} from '@patternfly/react-core';
+import { TimesIcon } from '@patternfly/react-icons';
+import { AppNotification, State } from '../redux/types';
+import { ackNotification, removeNotification } from '../redux/actions/actions';
+import { calculateRelativeTime } from '../utilities/utils';
+
+interface AppNotificationDrawerProps {
+  onClose: () => void;
+}
+
+const AppNotificationDrawer: React.FC<AppNotificationDrawerProps> = ({ onClose }) => {
+  const notifications: AppNotification[] = useSelector<State, AppNotification[]>(
+    (state) => state.appState.notifications,
+  );
+  const dispatch = useDispatch();
+  const newNotifications = React.useMemo(() => {
+    return notifications.filter((notification) => !notification.read).length;
+  }, [notifications]);
+  const [currentTime, setCurrentTime] = React.useState<Date>(new Date());
+
+  React.useEffect(() => {
+    const timeHandle = setInterval(() => setCurrentTime(new Date()), 20 * 1000);
```

Comment:
question: is `20 * 1000` for clarity/quick read of 20 secs? ( i personally like this format )

Reply:
correct (me too)